cmd/removeEnumerator.go

```diff
@@ -21,63 +21,43 @@
 package cmd
 
 import (
+	"bufio"
 	"context"
 	"errors"
 	"fmt"
+	"github.com/Azure/azure-pipeline-go/pipeline"
 	"github.com/Azure/azure-storage-azcopy/azbfs"
+	"github.com/Azure/azure-storage-azcopy/common"
 	"github.com/Azure/azure-storage-file-go/azfile"
 	"net/url"
+	"os"
 	"strings"
 )
 
-// provide an enumerator that lists a given blob resource (could be a blob or virtual dir)
+// provide an enumerator that lists a given resource (Blob, File)
 // and schedule delete transfers to remove them
-// TODO consider merging with newRemoveFileEnumerator
-func newRemoveBlobEnumerator(cca *cookedCopyCmdArgs) (enumerator *copyEnumerator, err error) {
-	sourceTraverser, err := newBlobTraverserForRemove(cca)
-	if err != nil {
-		return nil, err
-	}
+func newRemoveEnumerator(cca *cookedCopyCmdArgs) (enumerator *copyEnumerator, err error) {
+	var sourceTraverser resourceTraverser
 
-	// check if we are targeting a single blob
-	_, isSingleBlob := sourceTraverser.getPropertiesIfSingleBlob()
-
-	transferScheduler := newRemoveTransferProcessor(cca, NumOfFilesPerDispatchJobPart, isSingleBlob)
-	includeFilters := buildIncludeFilters(cca.includePatterns)
-	excludeFilters := buildExcludeFilters(cca.excludePatterns)
-
-	// set up the filters in the right order
-	filters := append(includeFilters, excludeFilters...)
-
-	finalize := func() error {
-		jobInitiated, err := transferScheduler.dispatchFinalPart()
+	if len(cca.listOfFilesLocation) > 0 {
```

Comment:
Can we detect, and throw a clear error, if it is a JSON file. Maybe just checking extension would do....

Reply:
I'm hesitant to rely on the extension as the file content might be in the right format despite the name. May be validate the first line is not "{"? I'm not sure.